src/test/java/business/AbstractSelectionControllerTests.java

```diff
@@ -188,6 +189,43 @@ void submitRequestForApproval() throws InterruptedException {
         SecurityContextHolder.clearContext();
     }
 
+    void submitRequestForApprovalWithNonApplicableAgreement() throws InterruptedException {
+        UserAuthenticationToken palga = getPalga();
+        SecurityContext securityContext = SecurityContextHolder.getContext();
+        securityContext.setAuthentication(palga);
+
+        RequestRepresentation representation =
+                requestController.getRequestById(palga, processInstanceId);
+        log.info("Status: " + representation.getStatus());
+
+        representation = requestController.claim(palga, processInstanceId, representation);
+
+        ((MockMailSender)mailSender).clear();
+
+        // only enforced in front end, not in back end
```

Comment:
This is indeed one of the targets for refactoring: moving `submitReview` to the `RequestService` and add validation, just like in `submit`.